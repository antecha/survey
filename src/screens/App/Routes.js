// @flow
import React from 'react';
import { Route } from 'react-router-dom';
import CreateSurvey from '../CreateSurvey';
import { StyledContent } from './styles';
import type { RoutesProps } from './types';
import type { Element } from './types';

export const AppContent = ({ children }: RoutesProps): Element<'StyledContent'> => (
  <StyledContent>{children}</StyledContent>
);

const Routes = (): Element<any> => {
  return (
    <React.Fragment>
      <Route exact path="/">
        <AppContent>Suvey-list</AppContent>
      </Route>

      <Route path="/create-survey">
        <AppContent>
          <CreateSurvey />
        </AppContent>
      </Route>
    </React.Fragment>
  );
};

export default Routes;
