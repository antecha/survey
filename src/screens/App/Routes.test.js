import React from 'react';
import { create } from 'react-test-renderer';
import Routes, { AppContent } from './Routes';
import { MemoryRouter } from 'react-router';

jest.mock('antd', _ => ({
  ...require.requireActual('antd'),
  Tabs: () => <div />,
}));

describe('AppContent', () => {
  const testRenderer = create(
    <AppContent>
      <div />
    </AppContent>,
  );

  it('renders', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe('Routes', () => {
  const testRenderer = create(
    <MemoryRouter>
      <Routes />,
    </MemoryRouter>,
  );

  it('renders', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
