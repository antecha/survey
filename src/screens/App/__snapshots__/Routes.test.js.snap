// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContent renders 1`] = `
<main
  className="sc-AxhUy hdwjou ant-layout-content"
>
  <div />
</main>
`;

exports[`Routes renders 1`] = `
Array [
  <main
    className="sc-AxhUy hdwjou ant-layout-content"
  >
    Suvey-list
  </main>,
  ",",
]
`;
